cleverhans/attacks.py

```diff
@@ -26,9 +26,17 @@ def __init__(self, model, back='tf', sess=None):
         if not(back == 'tf'):
             raise ValueError("Backend argument must either be 'tf'.")
 
-        if back == 'tf' and sess is None:
+        if back == 'tf':
```

Comment:
Can you comment on why did this check change?

Reply:
because ``` if back == 'tf'```
we want to set the variable self.tf_dtype = tf.as_dtype(dtypestr)

but sess might still be None or not None. Hence I split the if

```
        if back == 'tf':
            import tensorflow as tf
            self.tf_dtype = tf.as_dtype(dtypestr)
            if sess is None:
                sess = tf.get_default_session()
```